### ISSUE TITLE
Error macthing doesn't work with the `jsonrpcMessage` and `func (c *BoundContract) Transact`

### ISSUE THREAD

**Issue by krasi-georgiev** [ethereum/go-ethereum#22106]
#### System information
OS & Version: Linux
Commit hash : current master - f83fc302a504

I have some generated bindings and calling `func (c *BoundContract) Transact` returns an error 
```
<error(*github.com/ethereum/go-ethereum/rpc.jsonError)>)
Code: -32000
Message: "nonce too low"
Data: nil
```

Now the caller needs to match this against the `core.ErrNonceTooLow` , but since it is wrapped in `jsonError` this is not possible.

The end goal is to be able to use something like
```
if err == core.ErrNonceTooLow
	Nonce = Nonce + 1
or
if errors.Is(err, core.ErrNonceTooLow) {
	Nonce = Nonce + 1
```

I think the solution would be to expose `Is` method to `jsonError` and this will work nicely.

**Comment by MariusVanDerWijden** [comment]
Unfortunately it is not as easy as exposing the `Is` method on `jsonError` since the `jsonError` object does not have the error information. It only contains the error message not the actual error object. Since the `jsonError` has to be send over the wire it has to be marshaled into json. This will afaik result in loosing the error field if we were to put a raw `Error error` field into the `jsonError`. 
Currently the only working solution is to check the error string, which I agree is super brittle and shouldn't be done.
In theory we should use the error code to convey this information, but these are not standardized yet.
I'll convert this bug report into a feature request if that's cool with you?

**Comment by krasi-georgiev** [comment]
Yep, got it thanks for the update, matching by code is a good start. Can we maybe add this first for some common errors?
